internal/storage/runtime.go

```diff
@@ -187,21 +188,31 @@ func (r *runtimeService) createContainerOrPodSandbox(systemContext *types.System
 
 	// Pull out a copy of the image's configuration.
 	// Ideally we would call imageID.imageRef(r.storageImageServer), but storageImageServer does not have access to private data.
-	ref, err := istorage.Transport.NewStoreReference(r.storageImageServer.GetStore(), nil, template.imageID.privateID)
-	if err != nil {
-		return ContainerInfo{}, err
-	}
-	image, err := ref.NewImage(r.ctx, systemContext)
-	if err != nil {
-		return ContainerInfo{}, err
-	}
-	defer image.Close()
+	var imageConfig *v1.Image
+	if isRuntimePullImage {
+		var err error
+		artifact := ociartifact.New()
 
-	imageConfig, err := image.OCIConfig(r.ctx)
-	if err != nil {
-		return ContainerInfo{}, err
-	}
+		imageConfig, err = artifact.GetConfig(r.ctx, metadata.ImageName, &ociartifact.PullOptions{})
```

Comment:
Where do credentials for registry access come from? Can this work without them, only relying on the node-global cluster pull secret ??!